### ISSUE TITLE
JS SDK Spec

### ISSUE THREAD

**Issue by ericxtang** [livepeer/tutorial-app#1]
**Is your feature request related to a problem? Please describe.**
Create a Spec for the js sdk we talked about in the design session.  This sdk will be run as a daemon, and is used to interact with Livepeer nodes.  Features include:

* Job management (creation, termination)
* Token/Eth balance management
* Livepeer node management (we can start with a single node)
* **Value-add feature**: transmuxing video from webcam into RTMP, forward into Livepeer node

**Describe the solution you'd like**
An easy-to-use SDK for a javascript developer to create a serious video live streaming app with Livepeer.

**Additional context**
@jozanza, @ericxtang and @RaffiSapire had a discussion about this on 6/18/2018.

**Comment by dob** [comment]
If we do not have the "value add feature" of capturing camera video into RTMP in JS in the SDK, then are the other steps such as job management even worth it? What's the scenario that someone would create the job but not also stream video in the same stack? VOD where they have a pre-existing file?

**Comment by ericxtang** [comment]
@dob good point.  Our discussion was around creating an web application in js that will use Livepeer as the backend.  We thought it needed some logic to manage the jobs, token/eth balance, and the LP nodes it talks to, which can all fit into a js sdk (or client).

I agree that separately, we should also make sure the RESTFUL API for `go-livepeer` is well-structured.  I'll create a separate issue for it.
